sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallConnection.java

```diff
@@ -0,0 +1,227 @@
+     *
+     * @param audioFileUri The media resource uri of the play audio request. Currently only Wave file (.wav) format
+     *                     audio prompts are supported. More specifically, the audio content in the wave file must
+     *                     be mono (single-channel), 16-bit samples with a 16,000 (16KHz) sampling rate.
+     * @param playAudioOptions Options for play audio.
+     * @return the response payload for play audio operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public PlayAudioResult playAudio(String audioFileUri, PlayAudioOptions playAudioOptions) {
+        PlayAudioRequest playAudioRequest = PlayAudioConverter.convert(audioFileUri, playAudioOptions);
+        return callConnectionAsync.playAudio(playAudioRequest).block();
+    }
+
+    /**
+     * Play audio in a call.
+     *
+     * @param audioFileUri The media resource uri of the play audio request. Currently only Wave file (.wav) format
+     *                     audio prompts are supported. More specifically, the audio content in the wave file must
+     *                     be mono (single-channel), 16-bit samples with a 16,000 (16KHz) sampling rate.
+     * @param playAudioOptions Options for play audio.
+     * @param context A {@link Context} representing the request context.
+     * @return the response payload for play audio operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<PlayAudioResult> playAudioWithResponse(String audioFileUri,
+                                                             PlayAudioOptions playAudioOptions,
+                                                             Context context) {
+        PlayAudioRequest playAudioRequest = PlayAudioConverter.convert(audioFileUri, playAudioOptions);
+        return callConnectionAsync.playAudioWithResponse(playAudioRequest, context).block();
+    }
+
+    /**
+     * Disconnect the current caller in a Group-call or end a p2p-call.
+     *
+     * @return response for a successful Hangup request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Void hangup() {
+        return callConnectionAsync.hangup().block();
+    }
+
+    /**
+     * Disconnect the current caller in a Group-call or end a p2p-call.
+     *
+     * @param context A {@link Context} representing the request context.
+     * @return response for a successful HangupCall request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<Void> hangupWithResponse(Context context) {
+        return callConnectionAsync.hangupWithResponse(context).block();
+    }
+
+    /**
+     * Cancel all media operations in the call.
+     *
+     * @param operationContext operationContext.
+     * @return response for a successful CancelMediaOperations request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public CancelAllMediaOperationsResult cancelAllMediaOperations(String operationContext) {
```

Comment:
I am curious, what is this operationContext supposed to be and why is it type String? 
Wondering if you can add a better javadoc description for it: `* @param operationContext operationContext.`

Reply:
The operation context could be used in multiple ways. One way would be use a unique id as operation context for each operation and any events associated with that operation will have this id set. That way the client can co-relate them. Some times it is possible to get an operation related events even before the response to the request is received. 

Also operation context can be used to arbitrary encoded context value meaningful to the client. 

More description is added to the operationContext param description